### PR TITLE
Fix for confirmation email bug

### PR DESCRIPTION
**Pull request by stevelong00**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- This work is behind a feature toggle (flipper): YES
- Summary: Added logic to check the Pega status and not send a confirmation email when "not processed"
- To reproduce: Submit a test form in staging.  Then, hit the Pega status update endpoint with a call like this: `{'form_uuid': 'c1aabe64-08a8-4b33-951a-d914be8160a4', 'file_names': ['c1aabe64-08a8-4b33-951a-d914be8160a4_vha_10_7959f_2_supporting_doc-1.pdf'], 'status': 'Not Processed', 'case_id': ''}`.  Without this fix a confirmation email will be sent indicating a successful submit.  With this fix no email will be sent.
- This PR is the solution because it adds a check of the status and uses it to determine whether to send a confirmation email.
- I work for the IVC Forms team and we own this component.
- The success criteria is not sending a confirmation email when a Pega status of "Not Processed" is reported.

## Related issue(s)

- [Ticket 95556](https://github.com/department-of-veterans-affairs/va.gov-team/issues/95556)
- No epic

## Testing done

- [x] New code is covered by unit tests
- Without this fix a confirmation email will be sent when it should not be.
- Local testing is limited to unit tests.  Integration testing will have to be done in Staging.
- Testing plan:
  - Submit a test form in staging.  Then, hit the Pega status update endpoint with a call like this: `{'form_uuid': 'c1aabe64-08a8-4b33-951a-d914be8160a4', 'file_names': ['c1aabe64-08a8-4b33-951a-d914be8160a4_vha_10_7959f_2_supporting_doc-1.pdf'], 'status': 'Not Processed', 'case_id': ''}`.  Without this fix a confirmation email will be sent indicating a successful submit.  With this fix no email will be sent.

## Screenshots
None

## What areas of the site does it impact?
This only impacts the Pega status update endpoint.

## Acceptance criteria

- [x]  I added unit tests ~~and integration tests~~ for each feature
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback
None

### CODE REVIEW

**Review by michaelclement — DISMISSED** [review]
Looks good!

**Review by rjohnson2011 — DISMISSED** [review]

**Review by michaelclement — DISMISSED** [review]

File: modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb [hunk 1]
```diff
@@ -39,6 +39,8 @@ def update_status
 
       private
 
+      # rubocop:disable Metrics/MethodLength
```
Comment (ericboehs) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:42]: Could you disable using `.rubocop_todo.yml` instead?

You can run `rubocop --auto-gen-config --exclude-limit 100`.

Comment (stevelong00) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb]: @ericboehs  Done, but now rubocop does not seem to be picking up the exclusion.  What did I miss?

Comment (ericboehs) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb]: This is borked! Looks like a bug in rubicop :(

Go ahead and revert the rubocop_todo.yml change. Feels free to nuke the commit:

```
git checkout CHAMPVA-95556-confirmation-email-bug
git reset --hard HEAD~1
git push --force-with-lease
```

File: modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb [hunk 2]
```diff
@@ -52,7 +54,16 @@ def update_data(form_uuid, file_names, status, case_id)
```
Comment (ericboehs) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:54]: Alternatively you could collapse the above 6 lines into one:

```ruby
ivc_forms.each { |form| form.update!(pega_status: status, case_id:) }
```

**Review by ericboehs — DISMISSED** [review]

**Comment by ericboehs** [comment]
Hmmm. Now Champ VA upload specs are failing:

Screenshot 2024-12-11 at 10 41 07@2x

File: modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb (hunk 2 above)
Comment (stevelong00) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:47]: Done

**Review by michaelclement — APPROVED** [review]

File: modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb (hunk 2 above)
Comment (ericboehs) [reply modules/ivc_champva/app/controllers/ivc_champva/v1/pega_controller.rb:47]: Nice!

**Review by ericboehs — APPROVED** [review]

**Review by cloudmagic80 — APPROVED** [review]

**Pull request merged**